docs/src/rules/no-use-before-define.md

```diff
@@ -350,3 +362,106 @@ const d = 1;
 ```
 
 :::
+
+### enums (TypeScript only)
+
+Examples of **incorrect** code for the `{ "enums": true }` option:
+
+::: incorrect
+
+```ts
+/*eslint no-use-before-define: ["error", { "enums": true }]*/
+
+const x = Foo.FOO;
+
+enum Foo {
+  FOO,
+}
+```
+
+:::
+
+Examples of **correct** code for the `{ "enums": true }` option:
+
+::: correct
+
+```ts
+/*eslint no-use-before-define: ["error", { "enums": true }]*/
+
+enum Foo {
+  FOO,
+}
+
+const x = Foo.FOO;
+```
+
+:::
+
+### typedefs (TypeScript only)
+
+Examples of **incorrect** code for the `{ "enums": true }` option:
+
+::: incorrect
+
+```ts
+/*eslint no-use-before-define: ["error", { "typedefs": true }]*/
+
+let myVar: StringOrNumber;
+
+type StringOrNumber = string | number;
+
```

Comment:
Question: new `typedefs` option only reports the refs when `ignoreTypeReferences` is set to false and it is same for `typescript-eslint` also is it the expected behavior?

Reply:
Yes, it seems to be the expected behavior.

Reply:
Done!